### ISSUE TITLE
Abort Trap: 6 runtime error 

### ISSUE THREAD

**Issue by jez-nav** [nesbox/TIC-80#592]
After building on macOS 10.13.4, opening tic80.app does nothing but running in terminal produces the following message:

` TIC-80 tiny computer 0.70.0-dev.2
 http://tic.computer (C) 2017
Abort trap: 6
`

To reproduce: Be on macOS 10.13.4, Xcode 9.3, run "build macosx" then execute bin/tic80

During build, there are several warning messages pertaining to 3rd-party folder:
`ld: warning: object file (3rd-party/pre-built/macos/libSDL2.a(SDL_cocoametalview.o)) was built for newer OSX version (10.12) than being linked (10.6)`

Setting -mmacosx-version-min=10.13 removes the warning but the Abort trap: 6 still persists on runtime.

LLDB dump:
```
 TIC-80 tiny computer 0.70.0-dev.2
 http://tic.computer (C) 2017
2018-05-07 17:16:49.068104-0400 tic80[26984:1208136] detected buffer overflow
Process 26984 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x00007fff72805b6e libsystem_kernel.dylib`__pthread_kill + 10
libsystem_kernel.dylib`__pthread_kill:
->  0x7fff72805b6e <+10>: jae    0x7fff72805b78            ; <+20>
    0x7fff72805b70 <+12>: movq   %rax, %rdi
    0x7fff72805b73 <+15>: jmp    0x7fff727fcb00            ; cerror_nocancel
    0x7fff72805b78 <+20>: retq   
Target 0: (tic80) stopped.
```

**Comment by jez-nav** [comment]
So I found a workaround by adding `-D_FORTIFY_SOURCE=0` as a compiler option. It should disable the buffer overflow protection.

**Comment by nesbox** [comment]
fixed here 5e90d56
